### PR TITLE
Add methods for testing scalar sign and getting absolute value

### PR DESCRIPTION
**Pull request by isislovecruft**

### CODE REVIEW

File: src/scalar.rs [hunk 1]
```diff
@@ -496,6 +520,14 @@ impl UnpackedScalar {
         UnpackedScalar(*array_ref!(limbs,0,12))
     }
 
+    /// Determine if this `UnpackedScalar` is negative in constant time.
```
Comment (hdevalence) [inline src/scalar.rs]: What definition of negative does this use?

File: src/scalar.rs [hunk 2]
```diff
@@ -496,6 +520,14 @@ impl UnpackedScalar {
...
+    ///
+    /// # Returns
+    ///
+    /// If negative, return `-1i32`.  Otherwise, return `0i32`.
+    pub fn is_negative(&self) -> i32 {
```
Comment (hdevalence) [inline src/scalar.rs]: Should this use `u8`s like the rest of the CT code?

File: src/scalar.rs [hunk 3]
```diff
@@ -496,6 +520,14 @@ impl UnpackedScalar {
...
+        (*self[0] as i32) >> 255  // 255 = (sizeof(i32) * CHAR_BIT) - 1
```
Comment (hdevalence) [inline src/scalar.rs]: I think this should be 31, not 255

**Comment by isislovecruft** [comment]
Closed because there's no way any of this would be mergeable at this point.

**Pull request closed without merging**